ee-galleon-pack/src/main/resources/layers/standalone/batch-jberet/layer-spec.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="batch-jberet">
+    <dependencies>
+        <!-- CDI should be optional but to be conservative in the first cut at this, require it
+        <layer name="cdi" optional="true"/>
+        -->
+        <layer name="cdi"/>
+        <layer name="ee"/>
+        <layer name="transactions"/>
```

Comment:
@jamezp @chengfang @jfdenise 

A bit of a design/best practices question for you...

The batch-jberet subsystem supports a 'jdbc-repository' element which in turn requires that a datasource be configured.

I don't include any datasource related layer here because the 'batch-jberet' feature group is not using any 'jdbc-repository'.  If the user wanted to use that they would need to provision an actual DS (including the driver) so there's no benefit here to having an optional 'datasource' layer dep and always generating a bunch of config and pulling in other modules that are not used and are not even usable without further user effort.

I think I'm right but am pointing it out in case I'm wrong!

Another 'best practices' question is I declare a dep on 'ee' even though it is transitively required via 'transactions'.  But I see now I don't declare 'elytron' even though it is needed and is also pulled via transactions. My feeling is I should list them both. I think it's better practice to list non-obvious deps rather than relying on transitive ones. Particularly 'elytron', where one could imagine a future where someone would not require 'elytron' in 'transactions' and this would break.  I don't think it's necessary to spell out all deps everywhere (e.g. something the depends on jax-rs doesn't need to spell out a webserver dep), but for subtle cases it's probably better to be explicit.

Reply:
I agree with this. I don't see the need to the datasource, but Elytron does make sense.

I think EE also makes sense as there are modules from EE that I believe are used in deployment processing.

Reply:
I added elytron.